sympy/tensor/array/expressions/tests/test_array_expressions.py

```diff
@@ -677,3 +710,15 @@ def test_array_expr_construction_with_functions():
 
     expr = permutedims(PermuteDims(tp, [1, 0, 2, 3]), [0, 1, 3, 2])
     assert expr == PermuteDims(tp, [1, 0, 3, 2])
+
+
+def test_array_element_expressions():
```

Comment:
Should there be tests for symbolic elements too?

Reply:
There is one already in the derivatives. The one returning Kronecker deltas. Do you think we need more?

Reply:
I'm looking for the behaviour of e.g. `M[0,0].diff(Symbols('x'))`. For which I'm not even sure how to handle this as we don't know if the element depends on `x`, so perhaps it should never evaluate?

Reply:
`M[0, 0]` is a symbol which is different from `x`, so it should return zero. An array symbol is a symbol and its elements are symbols, not more complex expressions.

It could be more complicated if you have an `ArrayElement` object of a non-`ArraySymbol`. Say, you call `ArrayElement( ... some array or array expr ... , ... indices ... )`. In this case we should probably return some unevaluated expression.